gluoncv/model_zoo/rcnn/faster_rcnn/data_parallel.py

```diff
@@ -31,6 +31,8 @@ def __init__(self, net, optimizer, rpn_cls_loss, rpn_box_loss, rcnn_cls_loss, rc
                  mix_ratio, amp_enabled):
         super(ForwardBackwardTask, self).__init__()
         self.net = net
+        ctx = list(net.collect_params().values())[0].list_ctx()
+        self.net.target_generator.collect_params().reset_ctx(ctx)
```

Comment:
@chinakook This fix the issue since last PR where target_generator is a lambda function and the context cannot be set with `net.collect_params().reset_ctx`